benches/benches/vm_initialization.rs

```diff
@@ -75,36 +79,71 @@ fn transaction<R: Rng>(
 
 pub fn vm_initialization(c: &mut Criterion) {
     let mut rng = StdRng::seed_from_u64(8586);
-
+    let consensus_params = ConsensusParameters::default();
     let mut group = c.benchmark_group("vm_initialization");
 
-    let consensus_params = ConsensusParameters::default();
-    let mut i = 5usize;
-    loop {
+    // Increase the size of the script to measure the performance of the VM initialization
+    // with a large script. THe largest allowed script is 64 KB = 2^16 bytes.
+    const TX_SIZE_POWER_OF_TWO: usize = 16;
+
+    for i in 5..=TX_SIZE_POWER_OF_TWO {
         let size = 8 * (1 << i);
-        if size as u64 > consensus_params.script_params().max_script_data_length() {
-            break
-        }
+
         let script = vec![op::ret(1); size / Instruction::SIZE]
             .into_iter()
             .collect();
         let script_data = vec![255; size];
         let tx = transaction(&mut rng, script, script_data, &consensus_params);
         let tx_size = tx.transaction().size();
+        let tx = tx.test_into_ready();
+
         let name = format!("vm_initialization_with_tx_size_{}", tx_size);
         group.throughput(Throughput::Bytes(tx_size as u64));
         group.bench_function(name, |b| {
             b.iter(|| {
-                let mut vm = black_box(
+                let vm = black_box(
                     Interpreter::<_, Script, NotSupportedEcal>::with_memory_storage(),
                 );
-                let ready_tx = tx.clone().test_into_ready();
-                black_box(vm.init_script(ready_tx))
-                    .expect("Should be able to execute transaction");
+
+                // Initialize the VM and require the allocation of the whole memory
+                // to charge for the worst possible case.
+                black_box(initialize_vm(black_box(tx.clone()), vm));
```

Comment:
This is even more confusing from before. I'm not sure where the extra `black_box` came from either. 

Can we just have a method that says what we want, and then hide the implementation behind it? We probably can get rid of the comment in that case too.

`unoptimized_and_fully_saturated_vm` or something like that. Then have the internal code create the `vm` and wrap the `vm` and the `tx` in `black_box`?